src/lavinmq/auth/oauth_authenticator.cr

```diff
@@ -0,0 +1,268 @@
+require "./authenticator"
+require "./oauth_user"
+require "./../config"
+require "./jwt"
+require "./public_keys"
+require "./jwks_fetcher"
+require "openssl"
+require "http/client"
+require "./lib_crypto_ext"
+
+module LavinMQ
+  module Auth
+    module TokenVerifier
```

Comment:
This is only used for mocking? Why not monkey patch `OAuthAuthenticator` in the spec?

Reply:
`TokenVerifier` is also used in OAuthUser. OAuthAuthenticator has both authenticate and verify_token but OAuthUser doesn't need the whole Authenticator, only something that implementes TokenVerifier which could be something else.